threedi_schema/application/threedi_database.py

```diff
@@ -90,22 +59,27 @@ def base_path(self):
         return Path(self.path).absolute().parent
 
     def get_engine(self, get_seperate_engine=False):
+        # Ensure that path is a Path so checks below don't break
+        path = Path(self.path)
         if self._engine is None or get_seperate_engine:
-            if self.path == "":
+            if path == Path(""):
                 # Special case in-memory SQLite:
                 # https://docs.sqlalchemy.org/en/20/dialects/sqlite.html#threading-pooling-behavior
                 poolclass = None
             else:
                 poolclass = NullPool
-            engine = create_engine(
-                "sqlite:///{0}".format(self.path), echo=self.echo, poolclass=poolclass
-            )
-            listen(engine, "connect", load_spatialite)
+            if path.suffix.lower() == ".gpkg":
```

Comment:
FYI: if/else clauses can sometimes be rewritten to dictionary mappings. In this case it does not matter much because you don't expect more than 2 cases, but if that would be the case something like this might be an alternative:
```
engine_name, engine_fn = {
    ".gpkg": ("gpgk", load_spatialite_gpkg),
    ".sqlite": ("sqlite", load_spalitate)
}[path.suffix.lower()]
engine_path = f"{engine_name}:///{'' if path == Path('') else path}"
```

Note: I did not test this code... I just wanted to show it as an example ;)